websites/G/Gogoanime/presence.ts

```diff
@@ -1,120 +1,253 @@
-  var detail = "Browsing . . .";
-  var state: string;
-  const is404 = document.querySelector(
-    "#wrapper_bg > section > section.content_left > h1"
-  );
-  if (is404 != null && is404.textContent === "404") {
-    detail = "404";
-    state = "Non-existent page";
-  } else if (url === "/") {
-    state = "Recent animes release";
-  } else if (url === "/login.html") {
-    detail = "Logging in . . .";
-  } else if (url === "/register.html") {
-    detail = "Signing up . . .";
-  } else if (url === "/user/bookmark") {
-    detail = "Managing bookmarks";
-  } else if (url === "/anime-list.html") {
-    state = "Animes list";
-  } else if (url === "//search.html") {
-    detail = "Searching . . .";
-    const anime = decodeURI(document.location.href).split("=")[1].split(" ");
-    state = "Anime: " + formatAnime(anime);
-  } else if (url === "/new-season.html") {
-    state = "Newest animes";
-  } else if (url === "/anime-movies.html") {
-    state = "Animes' movies";
-  } else if (url === "/popular.html") {
-    state = "Popular animes";
-  } else if (url.includes("/genre/")) {
-    const genre = url.split("/").pop();
-    state = "Anime genre: " + upperCaseFirstChar(genre);
-  } else if (url.includes("/category/")) {
-    const anime = url.split("/").pop().split("-");
-    state = "Anime: " + formatAnime(anime);
-  } else {
-    detail = "Watching . . .";
-    let anime = url.split("/").pop().split("-");
-    const episode = anime[anime.length - 2] + " " + anime[anime.length - 1];
-    anime = anime.slice(0, anime.length - 2);
-    state = formatAnime(anime) + ": " + upperCaseFirstChar(episode);
-  }
-  var presenceData: PresenceData = {
-    largeImageKey: "logo",
-    details: detail,
-    state: state
-  };
-
-  if (detail === "Watching . . ." && videoInfos != null) {
-    if (
-      videoInfos.paused ||
-      (dataUpdated && videoInfos.currentTime === oldTime)
-    ) {
-      presenceData.smallImageKey = "pause";
-      presenceData.smallImageText =
-        getTimestampAsString(videoInfos.duration, videoInfos.currentTime) +
-        " left";
-      delete presenceData.endTimestamp;
-    } else {
-      presenceData.smallImageKey = "play";
-      presenceData.endTimestamp = getEndTime(
-        Math.floor(videoInfos.currentTime),
-        Math.floor(videoInfos.duration)
-      );
-      oldTime = videoInfos.currentTime;
-      dataUpdated = false;
-    }
-    presence.setActivity(presenceData, !videoInfos.paused);
-  } else {
-    presenceData.smallImageKey = "browsing";
-    presenceData.startTimestamp = new Date().getTime();
-    presence.setActivity(presenceData);
-  }
-});
+presence.on("UpdateData", async () => {
+  if (!isDomainChecked) {
+    isDomainChecked = true;
+    await checkDomain().then(result => {
+      isClone = result; 
+      console.log(
+        `[Premid Gogoganime Presence] The following gogoanime domain is ${result ? "a clone therefore not supported" : "supported"} by this extension.`);
+      });
```

Comment:
use presence.info or presence.error (and you spelled PreMiD wrong :triggered:)

Reply:
Annnd this is why I should read the docs more often xd Thx